code/game/machinery/vending.dm

```diff
@@ -761,6 +761,12 @@
 	if(emagged)
 		overlays += overlays_vending[2]
 
+/obj/machinery/vending/wirejack(var/mob/living/silicon/pai/P)
+	..()

```

Comment:
This is going to call the wirejack result even if the parent fails because hacking fails.

I suggest having the base wirejack return 1 if everything goes through, then doing if(..()) here